### PR TITLE
Fix flaky test in docker reader

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Fix flaky test in docker reader.

### CODE REVIEW

**Review by truthbk — APPROVED** [review]
Looks good! Thank you! 🙇

File: pkg/logs/input/docker/tailer_test.go [hunk 1]
```diff
@@ -119,7 +124,11 @@ func TestRead(t *testing.T) {
 
 func TestReadTimeout(t *testing.T) {
 	ctx, cancelFunc := context.WithCancel(context.Background())
-	tailer := NewTestTailer(&mockReaderSleep{ctx: ctx}, nil, cancelFunc)
+	reader := newMockReaderSleep(ctx)
+	// The reader should timout after testReadTimeout (10ms).
+	reader.timeout = 2 * time.Second
```
Comment (bmikaili) [inline pkg/logs/input/docker/tailer_test.go:129]: Why is this line necessary? Won't the reader just use `testReadTimeout regardless?`
